### PR TITLE
Abstract ElasticClient (auto-provide executor with .run)

### PR DESCRIPTION
**Pull request by arnoldlacko**

### CODE REVIEW

**Comment by arnoldlacko** [comment]
Closing, as this will be solved by adding a facade layer.

**Comment by arnoldlacko** [comment]
Reopening, as the discussion is still relevant. :)

File: modules/library/src/main/scala/zio/elasticsearch/ElasticClient.scala [hunk 1]
```diff
@@ -0,0 +1,11 @@
+package zio.elasticsearch
+
+import zio.{Task, ZLayer}
+
+abstract class ElasticClient(executor: ElasticExecutor) {
+
+  protected implicit class RunnableRequest[A](request: ElasticRequest[A]) {
+    def run: Task[A] = request.execute.provideLayer(ZLayer.succeed(executor))
```
Comment (mijicd) [inline modules/library/src/main/scala/zio/elasticsearch/ElasticClient.scala:8]: The downside of this approach is constant `provide`.

Comment (arnoldlacko) [reply modules/library/src/main/scala/zio/elasticsearch/ElasticClient.scala:8]: I was under the impression that having to `provide` is only a developer inconvenience, I missed the fact that there is a performance cost. I'll close this now.

**Pull request closed without merging**